ocs_ci/ocs/utils.py

```diff
@@ -1023,6 +1023,11 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False, status_failure=Tru
             raise ValueError(
                 f"must-gather fails in an disconnected environment bz-1974959\n{mg_output}"
             )
+        package_log_dir_path = os.path.join(log_dir_path, "rpm_package")
+        try:
```

Comment:
I would move the try block inside the function itself to wrap exact command on this line:

https://github.com/red-hat-storage/ocs-ci/pull/6058/files#diff-13cfcd3566979733be4860f4cc0db5af5382acbf7cf54fd94eb36fab45474c21R1301

container_output = ocp_obj.exec_oc_cmd(command)

As I can imagine the scenario that some pod you got info about at the start of the function will get deleted during the run. And there will not be such container anymore so you will loose information from everything.
But if it will just write warning inside the loop, it will try to continue to collect RPMs info from the rest of the pods/containers.

Reply:
Updated with suggested changes